packages/react-instantsearch-core/src/connectors/connectDynamicWidgets.ts

```diff
@@ -34,4 +39,6 @@ export default createConnector({
       attributesToRender: props.transformItems(facetOrder, { results }),
     };
   },
+
+  requiresInitialResults: true,
```

Comment:
Couldn't we use the existing Metadata API to flag a widget that needs initial results to render? Wondering if we could avoid transporting the connector description in the whole data flow.

Reply:
possible to put it in metadata, but that's used in current refinements, so I'll need to check a way to avoid it showing up there

Reply:
Why would it be a problem to have the flag exposed?

Reply:
because it's not a current refinement, so it shouldn't be an option in current refinements (that's what metadata is used for)